src/console_gui.cpp

```diff
@@ -68,9 +69,52 @@ static std::deque<IConsoleLine> _iconsole_buffer;
 
 static bool TruncateBuffer();
 
+class ConsoleAutoCompletion final : public AutoCompletion {
+public:
+	using AutoCompletion::AutoCompletion;
+
+private:
+	std::vector<std::string> GetSuggestions(std::string_view prefix, std::string_view query) override
+	{
+		while (prefix.ends_with(" ")) {
+			prefix.remove_suffix(1);
+		}
+
+		std::vector<std::string> suggestions;
+
+		// We only suggest commands or aliases, so we only do it for the first token or
+		// an argument to help command.
+		if (!prefix.empty() && prefix != "help") {
+			return {};
```

Comment:
Might as well just return `suggestions` ? (or move the declaration of the variable down)

Reply:
Done